go/vt/sqlparser/token.go

```diff
@@ -334,8 +334,16 @@ func (tkn *Tokenizer) Scan() (int, []byte) {
 		switch ch {
 		case eofChar:
 			return 0, nil
-		case '=', ',', ';', '(', ')', '+', '*', '%', '&', '|', '^', '~':
+		case '=', ',', ';', '(', ')', '+', '*', '%', '&', '^', '~':
 			return int(ch), nil
+		case '|':
+			switch tkn.lastChar {
```

Comment:
@bbeaudreault's suggestion looks good. The reasoning comes from the fact that go discourages indents. So, the switch statement would have to be written like this:
```
switch {
case '|':
  tkn.next()
  return OR, nil
}
return int(ch), nil
```

Then it becomes a switch statement with only one case. So, it's better off rewritten using `if`.